app/controllers/users_controller.rb

```diff
@@ -13,16 +22,58 @@ def new
   def create
     @user = User.new user_params
     if @user.save
-      flash[:success] = t :welcome
+      log_in @user
+      flash[:success] = t ".welcome"
       redirect_to @user
     else
       render :new
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      redirect_to root_path
+    else
+      if @user.update_attributes user_params
+        flash[:success] = t ".profile_updated"
+        redirect_to @user
+      else
+        render :edit
+      end
+    end
+  end
+
+  def destroy
+    User.find_by(id: params[:id]).destroy
+    flash[:success] = t ".user_deleted"
+    redirect_to users_url
+  end
+
   private
 
   def user_params
     params.require(:user).permit :name, :email, :password, :password_confirmation
   end
+
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = t ".please_login"
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
+    @user = User.find_by id: params[:id]
+    redirect_to root_url unless current_user? @user
+  end
+
+  def admin_user
```

Comment:
Bad naming! Name a function in more precise way depending on it's functionality. 
For example, you can name it verify_admin, verify_admin_user etc